### PR TITLE
MySql: Fix backticked identifier wrongly being reported as not found

### PR DESCRIPTION
**Pull request by pprkut**
### Brief summary of the change made

Fixes the MySQL part of #5356

### Pull Request checklist
- [ ] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by pprkut** [comment]
I added tests to verify the issue, but I'm not sure they are the correct tests, as they pass both with and without my change. Pointers appreciated on where I could add more appropriate tests :slightly_smiling_face:

File: test/fixtures/dialects/mysql/back_quoted_identifier.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+SELECT * FROM `foo`.`bar`;
```
Comment (WittierDinosaur) [inline test/fixtures/dialects/mysql/back_quoted_identifier.sql:1]: These are parser tests, to test if your change behaves as expected, you need to test the rules. See here: https://github.com/sqlfluff/sqlfluff/tree/main/test/fixtures/rules/std_rule_cases

**Comment by pprkut** [comment]
I've added a couple more tests now and came to the conclusion that removing `trim_chars=("`",)` doesn't actually fix it.

The problem is that the aliases that get checked in https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/rules/references/RF01.py#L216 do not have the backticks trimmed. I think I traced this back all the way to `get_eventual_alias()` in https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/dialects/dialect_ansi.py#L1554, but I'm not sure where to take it from here.

Does this mean `dialect_mysql.py` would need a custom `get_eventual_alias()` that removes the backticks? Or am I on the wrong track here?

**Comment by pprkut** [comment]
I missed that #5726 seems to already have fixed this. The test case I had no longer fails with the latest version of sqlfluff

**Pull request closed without merging**